tests/basic/basic_test.go

```diff
@@ -24,30 +24,36 @@ var _ = BeforeSuite(func() {
 
 // This test performs basic test of starting an application and destroying it (along with storage)
 var _ = Describe("{SetupTeardown}", func() {
+	var contexts []*scheduler.Context
+
 	It("has to setup, validate and teardown apps", func() {
-		var contexts []*scheduler.Context
 		for i := 0; i < Inst().ScaleFactor; i++ {
-			contexts = append(contexts, ScheduleAndValidate(fmt.Sprintf("setupteardown-%d", i))...)
+			contexts = append(contexts, ScheduleApps(fmt.Sprintf("setupteardown-%d", i))...)
 		}
 
-		opts := make(map[string]bool)
-		opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+		ValidateApps(fmt.Sprintf("validate apps for %s", CurrentGinkgoTestDescription().TestText), contexts)
+	})
+	fmt.Println(len(contexts))
+
+	AfterEach(func() {
+		TearDownAfterEachSpec(contexts)
+	})
 
-		for _, ctx := range contexts {
-			TearDownContext(ctx, opts)
-		}
+	JustAfterEach(func() {
+		DescribeNamespaceJustAfterEachSpec(contexts)
```

Comment:
This looks like it is going to describe them regardless of failure. Can you confirm this?